src/projection/util.ts

```diff
@@ -13,47 +42,54 @@ export function toAreaRangeSeries(data: AreaRangeData, index: number): any[] {
             symbol: "circle",
         },
     };
-    // Zero's and sub zeros are not supported on a logarithmic scale, hence adjusting 0 to 1.
-    const fixZeros = (value: number) => (value === 0 ? 1 : value);
-    data.data.forEach((rangeData) => {
-        const seriesData: any[] = [];
-        const averageData: any[] = [];
-        for (var i = 0; i < timeSeries.length - 1; i++) {
-            seriesData.push([
-                timeSeries[i],
-                fixZeros(rangeData.lower.value[i]),
-                fixZeros(rangeData.upper.value[i]),
-            ]);
-            averageData.push([
-                timeSeries[i],
-                fixZeros(rangeData.average.value[i]),
-            ]);
-        }
-        const rangePlot = {
-            name: `${rangeData.lower.id} - ${rangeData.upper.id}`,
-            data: seriesData,
+
+    const percentiledata2 = data.percentileData.find(
+        (_) => _.percentile == "2.5"
+    )?.data;
+    const percentiledata25 = data.percentileData.find(
+        (_) => _.percentile == "25"
+    )?.data;
+    const percentiledata50 = data.percentileData.find(
+        (_) => _.percentile == "50"
+    )?.data;
+    const percentiledata75 = data.percentileData.find(
+        (_) => _.percentile == "75"
+    )?.data;
```

Comment:
this is a bit too hard coded. I'd sort the percentileData in ascending order of percentile, and in front-end I merge 1 and N-1 as arearange, 2 and N-2 as arearange .... if there is single one left, draw as line chart.

Reply:
i will handle this through another PR.